File: lib/math/sum.ts
export function sum(...args: any[]): number;

export function sum(...args: Array<number>): number;

export function sum(...args: Array<number> | number[]): number {
	const val = args.length > 0 && typeof args[0] === 'object' ? args[0] : args;
	let total = 0;

	for (let i = 0, iLen = val.length; i < iLen; i += 1) {
		if (typeof val[i] === 'number') {
			total += val[i];
		}
	}

	return total;
}

File: packages/python/qsu/math/sum.py
def sum(*args):
	val = args[0] if len(args) > 0 and isinstance(args[0], (list, tuple)) else args
	total = 0

	for i in range(len(val)):
		if isinstance(val[i], (int, float)) and not isinstance(val[i], bool):
			total += val[i]

	return total
